tst/functional/TestKeepsGoingUp.kt

```diff
@@ -52,12 +56,32 @@ class TestKeepsGoingUp {
         game.run()
 
         assertNotEquals(map.positionFor(actor), exitPosition)
-        assertThat(player.feedback, equalTo(HIT_WALL_EXCEPTION_MESSAGE))
+        assertTrue(player.feedback is HitWallException)
+    }
+
+    @Test
+    fun testNavigatesWallAndWins() {
+        val builder = WorldMapBuilder()
+                .load("""
+                    .e.
+                    .X.
+                    .s.
```

Comment:
how about a more complicated test? one where player needs to actually look around to find the shortest path otherwise he's dead dead dead